tools/tfuser/cmds_network.go

```diff
@@ -214,23 +394,311 @@ func isIn(l []uint, i uint) bool {
 	return false
 }
 
-func generatePeers(nodeID string, n pkg.Network) []pkg.Peer {
-	peers := make([]pkg.Peer, 0, len(n.NetResources))
+func hasIPv4(n pkg.NetResource) bool {
+	for _, pep := range n.PubEndpoints {
+		if pep.To4() != nil {
+			return true
+		}
+	}
+	return false
+}
+
+// This function assumes:
+// - that a hidden node has functioning IPv4
+// - that a public node ALWAYS has public IPv6, and OPTIONALLY public IPv4
+// - that any public endpoint on any node is actually reachable (i.e. no firewall
+//		blocking incoming traffic)
+func generatePeers(n *pkg.Network) error {
```

Comment:
This function is quite long an complex.
I would love to see tests for it so we can more easily validate it's behavior and lock it down.